docarray/array/base.py

```diff
@@ -0,0 +1,24 @@
+from abc import ABC
+from docarray.array.mixins import AllMixins
+
+
+class BaseDocumentArray(ABC):
+    def __init__(self, *args, storage: str = 'memory', **kwargs):
+        super().__init__()
+        self._init_storage(*args, **kwargs)
```

Comment:
Is  `_init_storage()` still necessary?

Reply:
yes

Reply:
oh im actually not so sure now, will wait for @alaeddine-13